File: lib/features/retry_policy/retry_policy.dart
import 'package:flutter/material.dart';

class RetryPolicy with ChangeNotifier {
  bool shouldRetry(String errorType) {
    // Lógica para determinar se deve tentar novamente com base no tipo de erro
    return errorType == 'timeout' || errorType == 'connection_error';
  }

  void handleFailure(String errorType) {
    // Lógica para lidar com falha imediata
    if (!shouldRetry(errorType)) {
      // Notificar o usuário sobre a falha
      print('Falha ao realizar a operação: ');
    }
  }
}

File: backend/src/hono/routes/retry_policy/retry_policy.ts
import { Context } from 'hono';

const retryPolicy = async (c: Context, errorType: string) => {
  // Lógica para determinar se deve tentar novamente com base no tipo de erro
  if (errorType === 'timeout' || errorType === 'connection_error') {
    // Tentar novamente
    return true;
  }
  // Falhar imediatamente
  return false;
};

const handleFailure = async (c: Context, errorType: string) => {
  if (!await retryPolicy(c, errorType)) {
    // Retornar mensagem de erro para o usuário em português brasileiro
    return c.json({ error: 'Falha ao realizar a operação' }, 500);
  }
};

export default retryPolicy;
